lib/wibox/widget/slider.lua

```diff
@@ -89,7 +89,10 @@ local slider = {mt={}}
 -- @propemits true false
 -- @propbeautiful
 
---- The cursor icon while grabbing the handle. The available cursor names are located in "/usr/include/X11/cursorfont.h" without the XC_ prefixes.
+--[[
+    The cursor icon while grabbing the handle.
+    The available cursor names are located in "/usr/include/X11/cursorfont.h" without the XC_ prefixes.
```

Comment:
Isn't https://github.com/awesomeWM/awesome/blob/master/docs/common/cursor_c.ldoc based on the same source (hence a better documentation than pointing a file on the system)?

Reply:
I think it's too big to include at each occurrence, though.
Maybe include it once at a central place like `wibox.cursor`, then mention that here.

Reply:
So do you mean I should make a whole new page in the document, then include those cursors there? Where should I make that file?

Also, where does the `cursor_c.ldoc` is even used in the documentation? I searched the compiled documents, and no page contained those names.

Reply:
No new file needed. `wibox.cursor` is a property like the ones you've edited so far: https://awesomewm.org/apidoc/popups_and_bars/wibox.html#cursor. Since that's also the value that `slider.handle_cursor` and other widgets change internally, it's a good fir to point to as a single place to provide more detailed information.

It's currently used at https://github.com/awesomeWM/awesome/blob/master/root.c#L440

Reply:
To clarify, do you think that a list of cursors names should be added in https://awesomewm.org/apidoc/popups_and_bars/wibox.html#cursor, then referenced wherever necessary?

Reply:
Yes. Though not just _a_ list, but `cursor_c.ldoc` specifically.

Reply:
I searched and asked, but still not sure how to include an ldoc file inside a lua file's ldoc comment. The `root.c` file that uses the ldoc table is a .c file and I don't know how to do the same in a .lua file.

Reply:
This is the comment `@DOC_cursor_c_COMMON@` that is replaced by the actual content of the file at build time.

Reply:
> This is the comment `@DOC_cursor_c_COMMON@` that is replaced by the actual content of the file at build time.
> 
> See https://github.com/awesomeWM/awesome/blob/master/docs%2Fload_ldoc.cmake

I used the same syntax in a lua file but it didn't work (`@DOC_cursor_c_COMMON@` is used in a C file)

Reply:
I found out why the @DOC_...@ strings were not getting replaced by their corresponding files. It was because I was building the docs directly from the `/docs` directory. Instead, I had to build awesome first, then build the docs from `/build/docs` . Now I build the docs by running `ldoc .` in `/build/docs`.

Reply:
I encountered another problem while including the cursor names in the slider widget docs. When I change

```lua
--- The cursor icon while grabbing the handle. The availabe cursor names are:
-- @property handle_cursor
-- @tparam[opt="fleur"] string|nil handle_cursor
-- @propbeautiful
-- @see mousegrabber
```
to
```lua
--- The cursor icon while grabbing the handle. The availabe cursor names are:
--
--@DOC_cursor_c_COMMON@
--
-- @property handle_cursor
-- @tparam[opt="fleur"] string|nil handle_cursor
-- @propbeautiful
-- @see mousegrabber
```
the section explaining handle_cursor completely disappears from the compiled docs, and I get this error while building the docs:
```
WARNING: 	beautiful.slider_handle_cursor	is not used by anything, add @usebeautiful or @propbeautiful
```

Reply:
The problem was because the lines in `cursor_c.ldoc` file began with '*' but for being included in another lua file, they needed to be started with '--'. I made a new file called `cursor_c_lua.ldoc` with new lines starting with `--`.